Build error with Unexpected token when importing react-native/tvos-types.d.ts

### Discussed in https://github.com/react-native-tvos/react-native-tvos/discussions/474

<div type='discussions-op-text'>

<sup>Originally posted by **matias-sandell** March 16, 2023</sup>
Trying out the new https://www.npmjs.com/package/react-native-tvos/v/0.69.8-0rc3 release and getting a build error when a TypeScript component imports `react-native/tvos-types.d`.

```
error SyntaxError: ......../node_modules/react-native/tvos-types.d.ts: Unexpected token (115:21)

  113 |   export type FocusGuideMethods = {
  114 |     setDestinations: (
> 115 |       destinations: (?React.ElementRef<HostComponent<unknown>>)[],
      |                      ^
  116 |     ) => void;
  117 |   }
  118 |.

FAILURE: Build failed with an exception.
```

<img width="981" alt="Screenshot 2023-03-16 at 11 16 03" src="https://user-images.githubusercontent.com/437760/225587679-0693afe5-66e4-4fc5-bc06-80f5b00662a9.png">

Workarounds
- Removing the `react-native/tvos-types.d` import from the .tsx file makes the build work again (and the app works when building for production, but not when running it through the react native dev server)
- Removing the `?` from the type definition also makes the app work through the dev server.

It's probably something with my setup but any hints as to what the underlying problem might be will be greatly appreciated! 🙌

The setDestinations type definition was introduced in https://github.com/react-native-tvos/react-native-tvos/commit/7143d7877ec432337f36693fba75f611178a96e8#diff-e14263f50370e82b314d0b19b9a21e70e0a0948798e28872703fdd1093b3610cR111</div>

--- a/tvos-types.d.ts
+++ b/tvos-types.d.ts
@@ -112,7 +112,7 @@ declare module 'react-native' {
 
   export type FocusGuideMethods = {
     setDestinations: (
-      destinations: (?React.ElementRef<HostComponent<unknown>>)[],
+      destinations: (React.ElementRef<HostComponent<unknown>> | null)[],
     ) => void;
   }
 
@@ -123,7 +123,7 @@ declare module 'react-native' {
    * An example is provided in `RNTester` that shows different ways of using this component.
    * https://github.com/react-native-tvos/react-native-tvos/blob/tvos-v0.69.8/packages/rn-tester/js/examples/TVFocusGuide/TVFocusGuideAutoFocusExample.js
    */
-  export const TVFocusGuideView: React.ForwardRefExoticComponent<FocusGuideProps & React.RefAttributes<FocusGuideMethods>>;
+  export const TVFocusGuideView: React.ForwardRefExoticComponent<FocusGuideProps & React.RefAttributes<View & FocusGuideMethods>>;
   export interface TVTextScrollViewProps extends ScrollViewProps {
     /**
      * The duration of the scroll animation when a swipe is detected.

--- a/tvos-types.d.ts
+++ b/tvos-types.d.ts
@@ -112,7 +112,7 @@ declare module 'react-native' {
 
   export type FocusGuideMethods = {
     setDestinations: (
-      destinations: (React.ElementRef<HostComponent<unknown>> | null)[],
+      destinations: (React.ElementRef<HostComponent<unknown>> | null | undefined)[],
     ) => void;
   }
 

